client/dashboard/me/active-subscriptions/active-subscription-description.tsx

```diff
@@ -0,0 +1,225 @@
+	if ( 'theme' === purchase.product_type ) {
+		return __( 'Premium Theme' );
+	}
+
+	if ( 'concierge-session' === purchase.product_slug ) {
+		return __( 'One-on-one Support' );
+	}
+
+	if ( purchase.partner_name ) {
+		if ( purchase.partner_type && [ 'agency', 'a4a_agency' ].includes( purchase.partner_type ) ) {
+			return __( 'Agency Managed Plan' );
+		}
+
+		return __( 'Host Managed Plan' );
+	}
+
+	if ( purchase.is_plan ) {
+		return __( 'Site Plan' );
+	}
+
+	if ( purchase.is_domain_registration ) {
+		return purchase.product_name;
+	}
+
+	if ( purchase.product_slug === 'domain_map' ) {
+		return purchase.product_name;
+	}
+
+	if ( isTemporarySitePurchase( purchase ) && purchase.product_type === 'akismet' ) {
+		return null;
+	}
+
+	if ( isTemporarySitePurchase( purchase ) && purchase.product_type === 'saas_plugin' ) {
+		return null;
+	}
+
+	if ( isTemporarySitePurchase( purchase ) && isA4ATemporarySitePurchase( purchase ) ) {
+		return null;
+	}
+
+	if ( purchase.is_google_workspace_product && purchase.meta ) {
+		return sprintf(
+			// translators: The domain is the domain name of the site
+			__( 'Mailboxes and Productivity Tools at %(domain)s' ),
+			{
+				domain: purchase.meta,
+			}
+		);
+	}
+
+	if ( purchase.is_titan_mail_product && purchase.meta ) {
+		return sprintf(
+			// translators: The domain is the domain name of the site
+			__( 'Mailboxes at %(domain)s' ),
+			{
+				domain: purchase.meta,
+			}
+		);
+	}
+
+	if ( purchase.product_type === 'marketplace_plugin' || purchase.product_type === 'saas_plugin' ) {
+		return __( 'Plugin' );
+	}
+
+	if ( purchase.meta ) {
+		return purchase.meta;
+	}
+
+	return null;
+}
+
+export function ActiveSubscriptionDescription( {
+	purchase,
+	site,
+	getUrlForSiteLevelView,
+}: {
+	purchase: ActiveSubscription;
+	site?: { name: string; slug: string; ID: number };
+	getUrlForSiteLevelView: ( siteId: number ) => string;
+} ) {
+	if ( isTemporarySitePurchase( purchase ) ) {
+		return null;
+	}
+
+	const productType = purchaseType( purchase );
+
+	if ( site ) {
+		if ( productType && site.name && site.slug ) {
+			return (
+				<div>
+					{ createInterpolateElement(
+						sprintf(
+							// translators: The string contains the product name, the name of the site, and the URL for the site e.g. Premium plan for Block Store (blockstore.com)
+							__(
+								'%(purchaseType)s for <button>%(siteName)s</button> (<link>%(siteDomain)s</link>)'
+							),
+							{
+								purchaseType: productType,
+								siteName: site.name,
+								siteDomain: site.slug,
+							}
+						),
+						{
+							button: (
+								<Link
+									to={ getUrlForSiteLevelView( site.ID ) }
+									title={
+										// translators: the siteName is the name of the site
+										sprintf( __( 'View subscriptions for %(siteName)s' ), {
+											siteName: site.name,
+										} )
+									}
+								/>
+							),
+							link: (
+								<a
```

Comment:
Team review: [nit] Consider using ExternalLink for consistency on external links handling.

Reply:
This doesn't work when combined with `createInterpolateElement()` in TypeScript. It can't tell that the `children` prop will be added later.

```
Property 'children' is missing in type '{ href: string; rel: string; title: string; }' but required in type 'Omit<WordPressComponentProps<ExternalLinkProps, "a", false>, "target">'.
```